compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -346,6 +346,11 @@ impl<'context> Elaborator<'context> {
             &items.module_attributes,
         );
 
+        // Make sure trait impl functions have the same visibility as their traits
+        for trait_impl in &items.trait_impls {
+            self.set_trait_impl_functions_visibility(trait_impl);
+        }
```

Comment:
Does this need to be a separate loop? Can we move it into `collect_trait_impl`?

Reply:
Done! I didn't realize we had two loops for trait impls.